test/unittest/dds/subscriber/DataReaderTests.cpp

```diff
@@ -845,6 +845,218 @@ TEST_F(DataReaderTests, resource_limits)
     }
 }
 
+void check_sample(
+        const SampleInfo& info_1,
+        const SampleInfo& info_2,
+        const FooType& data_1,
+        const FooType& data_2,
+        bool should_be_equal)
+{
+    EXPECT_EQ(should_be_equal, info_1.sample_identity == info_2.sample_identity);
+    EXPECT_EQ(should_be_equal, data_1 == data_2);
+}
+
+void check_sample_values(
+        const FooSeq& data,
+        const std::string& values)
+{
+    EXPECT_EQ(data.length(), values.size());
+
+    for (FooSeq::size_type i = 0; i < data.length(); ++i)
+    {
+        EXPECT_EQ(values[i], data[i].message()[0]);
+    }
+}
+
+/*
+ * This test checks that the behaviour of the sample_states parameter of read/take calls.
```

Comment:
```suggestion
 * This test checks that the behaviour of the sample_states parameter of read/take calls.
 * A writer sends num_samples messages, and the reader execute the following:
 *  1. Try to read/take already read msg : None.
 *  2. Read in 6 different ways: read 2 first msgs. <could be explain in a schema, maybe with compare_cases structure
 *  3. Take msgs in different ways. <these are already well explained one by one along the code>
```

In my opinion this is a complex test that probes complicated and different functionality, so it should be better commented or divided in two different tests: read & take.

Reply:
I improved the inline doc a bit on 4c5145a